compiler/rustc_ty_utils/src/needs_drop.rs

```diff
@@ -179,6 +192,43 @@ fn adt_drop_tys(tcx: TyCtxt<'_>, def_id: DefId) -> Result<&ty::List<Ty<'_>>, Alw
     res.map(|components| tcx.intern_type_list(&components))
 }
 
+fn adt_significant_drop_tys(
+    tcx: TyCtxt<'_>,
+    def_id: DefId,
+) -> Result<&ty::List<Ty<'_>>, AlwaysRequiresDrop> {
```

Comment:
can we refactor this fn body into a helper fn that is shared by `adt_drop_tys` and `adt_significant_drop_tys`? they seem like virtually the same fn

Reply:
done